### PR TITLE
Improve production bundle

### PR DESCRIPTION
**Pull request by thomasdigby**
Re-organised webpack config, updated to webpack 2, added some production plugins, added `stats.json` for analysing bundle output.

### CODE REVIEW

**Comment by thomasdigby** [comment]
This requires node >= 4.7 FYI.

**Comment by thomasdigby** [comment]
[Create production config for client script](https://trello.com/c/wn1wuNo6/27-create-production-config-for-client-script)

File: src/webpack.config.js [hunk 1]
```diff
@@ -1,38 +1,78 @@
...
-        // new webpack.DefinePlugin({
-        //   'process.env': {
-        //     'NODE_ENV': JSON.stringify('production')
-        //   }
-        // })
-      ]
+const env = process.env.NODE_ENV
+
+
+export default (ctx, isTree = false) => {
```
Comment (jonsherrard) [inline src/webpack.config.js]: Do we use isTree anymore?

Comment (thomasdigby) [inline src/webpack.config.js]: Yep, https://github.com/shortlist-digital/tapestry-wp/pull/25/files#diff-6d5e756cc942fd5f521d854dbb7bfe3bR33

**Comment by jonsherrard** [comment]
If it's `4.7` only shall we put an engines field into package.json http://www.marcusoft.net/2015/03/packagejson-and-engines-and-enginestrict.html

File: src/webpack.config.js (hunk 1 above)
Comment (jonsherrard) [inline src/webpack.config.js]: I don't think that ever gets run at the moment, we just compile the whole folder: https://github.com/shortlist-digital/tapestry-wp/blob/develop/bin/tapestry#L17 BUT, let's leave it in, as we might go back to this way of doing it I think.

File: src/webpack.config.js [hunk 2]
```diff
@@ -1,38 +1,78 @@
...
+    resolve: {
+      modules: [ctx, `${ctx}/node_modules`]
+    },
+    module: {
+      rules: [{
+        test: /\.js$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        options: {
+          presets: ['es2015', 'react']
```
Comment (mathewhawley) [inline src/webpack.config.js]: I'm not 100% on how the 'tree-shaking' in Webpack 2 works, but if you want to take advantage, I think you may need to specify 'modules: false' for the es2015 preset? I could be wrong, but I've glanced over a few set ups and articles that say something along those lines.

Or this guy – the article is from early last year though (https://medium.com/modus-create-front-end-development/webpack-2-tree-shaking-configuration-9f1de90f3233#.3knrqfiz1) – who says `es2015-native-modules` will help enable 'tree-shaking'

Comment (thomasdigby) [inline src/webpack.config.js]: Yeah I've had a few plays with this, I'll try again on a separate branch, maybe we can pair-program it next week?

**Comment by thomasdigby** [comment]
@mathewhawley Will this impact your CLI work at all? This prod webpack config might be superseded by the possibility of merging webpack & babel configs at some point.

**Pull request merged**